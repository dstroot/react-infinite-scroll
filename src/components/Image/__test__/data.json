{
  "id": "Ic-NitASHiI",
  "created_at": "2020-04-04T09:38:43-04:00",
  "updated_at": "2020-04-04T11:39:02-04:00",
  "promoted_at": "2020-04-04T11:39:02-04:00",
  "width": 4000,
  "height": 6000,
  "color": "#D7E0EF",
  "description": "Filmmaking",
  "alt_description": "man in blue polo shirt sitting on chair near window",
  "urls": {
    "raw": "https://images.unsplash.com/photo-1586007484979-bbea94d7fae0?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjk5NjQ5fQ",
    "full": "https://images.unsplash.com/photo-1586007484979-bbea94d7fae0?ixlib=rb-1.2.1&q=85&fm=jpg&crop=entropy&cs=srgb&ixid=eyJhcHBfaWQiOjk5NjQ5fQ",
    "regular": "https://images.unsplash.com/photo-1586007484979-bbea94d7fae0?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjk5NjQ5fQ",
    "small": "https://images.unsplash.com/photo-1586007484979-bbea94d7fae0?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjk5NjQ5fQ",
    "thumb": "https://images.unsplash.com/photo-1586007484979-bbea94d7fae0?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max&ixid=eyJhcHBfaWQiOjk5NjQ5fQ"
  },
  "links": {
    "self": "https://api.unsplash.com/photos/Ic-NitASHiI",
    "html": "https://unsplash.com/photos/Ic-NitASHiI",
    "download": "https://unsplash.com/photos/Ic-NitASHiI/download",
    "download_location": "https://api.unsplash.com/photos/Ic-NitASHiI/download"
  },
  "categories": [],
  "likes": 1,
  "liked_by_user": false,
  "current_user_collections": [],
  "user": {
    "id": "G__U81B4QDY",
    "updated_at": "2020-04-04T11:39:02-04:00",
    "username": "castillcc",
    "name": "cristian castillo",
    "first_name": "cristian",
    "last_name": "castillo",
    "twitter_username": "_castill_",
    "portfolio_url": "http://instagram.com/casti.iam/",
    "bio": "Siempre traigo una c√°mara conmigo.",
    "location": "Santiago de Chile",
    "links": {
      "self": "https://api.unsplash.com/users/castillcc",
      "html": "https://unsplash.com/@castillcc",
      "photos": "https://api.unsplash.com/users/castillcc/photos",
      "likes": "https://api.unsplash.com/users/castillcc/likes",
      "portfolio": "https://api.unsplash.com/users/castillcc/portfolio",
      "following": "https://api.unsplash.com/users/castillcc/following",
      "followers": "https://api.unsplash.com/users/castillcc/followers"
    },
    "profile_image": {
      "small": "https://images.unsplash.com/profile-1578097758155-7e75f9c1de32image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
      "medium": "https://images.unsplash.com/profile-1578097758155-7e75f9c1de32image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
      "large": "https://images.unsplash.com/profile-1578097758155-7e75f9c1de32image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128"
    },
    "instagram_username": "Casti.iam",
    "total_collections": 0,
    "total_likes": 51,
    "total_photos": 221,
    "accepted_tos": true
  }
}
