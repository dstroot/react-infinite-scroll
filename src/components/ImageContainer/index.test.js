import React from "react";
import { render } from "@testing-library/react";

// component to test
import { ImageContainer } from ".";
import data from "./__test__/data.json";

describe("ImageContainer", () => {
  const windowIntersectionObserver = window.IntersectionObserver;
  const visibleMock = jest.fn();

  beforeAll(() => {
    const mockEntry = { isIntersecting: true };

    window.IntersectionObserver = jest.fn(function () {
      this.observe = () => {};
      this.unobserve = jest.fn();
    });
  });

  afterAll(() => {
    window.IntersectionObserver = windowIntersectionObserver;
  });

  it("renders", () => {
    // arrange
    const { asFragment } = render(
      <ImageContainer
        src={data.urls.regular}
        thumb={data.urls.thumb}
        height={data.height}
        width={data.width}
        alt={data.alt_description}
        url={data.links.html}
        onIsVisible={visibleMock}
      />
    );

    // act

    // assert
    expect(asFragment()).not.toBeNull();
    expect(asFragment()).toMatchSnapshot();
    expect(visibleMock.mock.calls.length).toBe(0);
  });
});

// Testing Intersection Observers
// https://gist.github.com/ianmcnally/4b68c56900a20840b6ca840e2403771c
