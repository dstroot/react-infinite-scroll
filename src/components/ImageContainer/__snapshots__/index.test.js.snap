// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageContainer renders 1`] = `
<DocumentFragment>
  <a
    class="image-container"
    href="https://unsplash.com/photos/Ic-NitASHiI"
    rel="noopener noreferrer"
    style="padding-bottom: 150%;"
    target="_BLANK"
  />
</DocumentFragment>
`;
